k-distribution/include/kframework/builtin/domains.md

```diff
@@ -465,6 +465,230 @@ module MAP-SYMBOLIC
 endmodule
 ```
 
+Range maps
+----
+
+Provided here is the syntax of an implementation of immutable, associative,
+commutative range maps from `Int` to `KItem`. This type is hooked to an
+implementation of range maps provided by the backend.
+Although the underlying range map data structure supports any key sort, the
+current implementation by the backend only supports `Int` keys due to
+limitations of the underlying ordering function.
```

Comment:
Is this supported by the haskell backend as well? If not, then I think it should be mentioned here that only the llvm backend implements this feature.

I think we also have some module attributes to protect against wrongful inclusion.

Reply:
In a follow-up PR, we can probably make an inefficient version of the RangeMap sort written entirely in K for use by the Haskell backend. I don't see the need to adjust the module attributes currently, although I agree about the documentation.